README.md

```diff
@@ -1,26 +1,97 @@
-<!--
-Avoid using this README file for information that is maintained or published elsewhere, e.g.:
+<div align="center">
 
-* metadata.yaml > published on Charmhub
-* documentation > published on (or linked to from) Charmhub
-* detailed contribution guide > documentation or CONTRIBUTING.md
+# CephFS server proxy operator
 
-Use links instead.
--->
+A [Juju](https://juju.is) operator for proxying exported Ceph filesystems.
 
-# cephfs-proxy-operator
+[![Charmhub Badge](https://charmhub.io/cephfs-server-proxy/badge.svg)](https://charmhub.io/cephfs-server-proxy)
+[![CI](https://github.com/canonical/cephfs-server-proxy-operator/actions/workflows/ci.yaml/badge.svg)](https://github.com/canonical/cephfs-server-proxy-operator/actions/workflows/ci.yaml/badge.svg)
+[![Release](https://github.com/canonical/cephfs-server-proxy-operator/actions/workflows/release.yaml/badge.svg)](https://github.com/canonical/cephfs-client-operator/actions/workflows/release.yaml/badge.svg)
+[![Matrix](https://img.shields.io/matrix/ubuntu-hpc%3Amatrix.org?logo=matrix&label=ubuntu-hpc)](https://matrix.to/#/#ubuntu-hpc:matrix.org)
 
-Charmhub package name: operator-template
-More information: https://charmhub.io/cephfs-proxy-operator
+</div>
 
-Describe your charm in one or two sentences.
+## Features
 
-## Other resources
+The CephFS server proxy operator enable CephFS client operators to mount exported Ceph filesystems
+on Ceph clusters not managed by Juju. This proxy operator enables Juju users to manually set 
+a CephFS share endpoint that their CephFS client operators need to mount.
 
-<!-- If your charm is documented somewhere else other than Charmhub, provide a link separately. -->
+## Usage
 
-- [Read more](https://example.com)
+#### With a minimal CephFS kernel server
```

Comment:
Hmm... not sure this is the right terminology to use here :thinking: 

"kernel server" is used in the NFS operator description because we're using the `nfs-kernel-server` package from the Ubuntu package archive - `apt info nfs-kernel-server`. IIRC it's called "kernel server" because the server runs in kernel mode rather than in user space where there's more restrictions.

Perhaps `With microceph` would be better?

```suggestion
#### With microceph
```